src/components/screens/all-software-screen.tsx

```diff
@@ -0,0 +1,505 @@
+
+    const matchesTerms = searchTerms.every(term => {
+      const lowerCaseTerm = term.toLowerCase();
+      return (
+        app.name.toLowerCase().includes(lowerCaseTerm) ||
+        (app.tags ?? []).some(tag => tag.toLowerCase().includes(lowerCaseTerm))
+      );
+    });
+
+    return matchesInput && matchesTerms;
+  };
+
+  const filterByStatus = (app: SoftwareRegistry, status: string): boolean => {
+    return status === "ALL" || app.status === status;
+  };
+
+  const filteredApplications = useMemo(() => {
+    return applications.filter(app => {
+      const matchesSearch = filterBySearchTerms(app);
+      const matchesStatus = filterByStatus(app, selectedStatus);
+      return matchesSearch && matchesStatus;
+    });
+  }, [applications, inputValue, searchTerms, selectedStatus]);
+
+  /**
+   * Handles the input for search terms.
+   * Adds new search terms when the "Enter" key is pressed.
+   */
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === "Enter" && inputValue.trim() !== "") {
+      event.preventDefault();
+
+      if (!searchTerms.includes(inputValue.trim())) {
+        setSearchTerms([...searchTerms, inputValue.trim()]);
+        setInputValue("");
+      }
+    }
+  };
+
+  /**
+   * Deletes a search term chip.
+   */
+  const handleDeleteChip = (chipToDelete: string) => {
+    setSearchTerms((prevChips) =>
+      prevChips.filter((chip) => chip !== chipToDelete)
+    );
+  };
+
+  /**
+   * Updates the status of application.
+   * 
+   * @param {string} id - The id of the application to update.
+   * @param {SoftwareStatus} newStatus - The new status to assign to the application.
+   */
+  const handleStatusChange = async (id: string, newStatus: SoftwareStatus) => {
+    try {
+      const appToUpdate = applications.find(app => app.id === id);
+
+      if (appToUpdate) {
+        const updatedApp: SoftwareRegistry = {
+          ...appToUpdate,
+          status: newStatus,
+          name: appToUpdate.name || "",
+          description: appToUpdate.description || "",
+          review: appToUpdate.review || "",
+          url: appToUpdate.url || "",
+          image: appToUpdate.image || "",
+          tags: appToUpdate.tags || [],
+          users: appToUpdate.users || []
+        };
+
+        const updatedApplications = applications.map(app =>
+          app.id === id ? updatedApp : app
+        );
+        setApplications(updatedApplications);
+
+        await softwareApi.updateSoftwareById({
+          id,
+          softwareRegistry: updatedApp
+        });
+      }
+    } catch (error) {
+      console.error(`Error updating status: ${error}`);
+    }
+  };
+
+  /**
+   * Adds the current user to the list of users for the specified application.
+   * 
+   * @param {string} id - The id of the application to save.
+   */
+  const handleSave = async (id: string) => {
+    try {
+      const applicationToUpdate = applications.find(app => app.id === id);
```

Comment:
software not application- please check thorughout the code base for this renaming